docs/06-concepts/18-testing/02-the-basics.md

```diff
@@ -0,0 +1,287 @@
+# The basics
+
+## Using `sessionBuilder` to set up a test scenario
+
+The `withServerpod` helper provides a `sessionBuilder` object that helps with setting up different scenarios for tests. It looks like the following:
+
+```dart
+/// A test specific builder to create a [Session] that for instance can be used to call database methods.
+/// The builder can also be passed to endpoint calls. The builder will create a new session for each call.
+abstract class TestSessionBuilder {
+  /// Given the properties set on the session through the `copyWith` method,
+  /// this returns a serverpod [Session] that has the configured state.
+  Session build();
+
+  /// Creates a new unique session with the provided properties.
+  /// This is useful for setting up different session states in the tests
+  /// or simulating multiple users.
+  TestSessionBuilder copyWith({
+    AuthenticationOverride? authentication,
+    bool? enableLogging,
+  });
+}
+```
+
+To create a new state, simply call `copyWith` with the new properties and use the new session builder in the endpoint call.
+
+### Setting authenticated state
+
+To control the authenticated state of the session, the `AuthenticationOverride` class can be used:
+
+```dart
```

Comment:
Suggestion/Discussion: Don't expose the implementation details of the TestSessionBuilder in the docs.

The implementation details for the TestSessionBuilder are going to evolve over time. Will we keep up with the documentation? Maybe we can explain it more abstract and then go into single details for each "copyWith" capability instead?

Reply:
I would argue that this is not implementation details as I'm only showing the public interface (without any implementation code). However we phrase this part - if the public interface changes that will inevitable require us to sync with the docs.

That said, maybe I can dissect this an lift out the comments as regular documentation text and then have the functions as snippets. E.g. something along these lines: 
>  `copyWith` creates a new unique session with the provided properties. This is useful for setting up different session states in the tests or simulating multiple users.
```dart
  TestSessionBuilder copyWith({
    AuthenticationOverride? authentication,
    bool? enableLogging,
  });
```

Reply:
You are correct, this is the interface, but the comments we have there and the structure of the code is implementation details. So I think it makes more sense to highlight the specific interfaces and keep away from documenting class declarations.

Reply:
Tried a rework here, but it's a bit tricky. Let me know what you think: 049c08d